lib/errors.js

```diff
@@ -32,9 +32,10 @@ module.exports = {
         this.name = 'UNREGISTRATION_ERROR';
         this.message = 'Error unregistering context provider for device: ' + id + ' of type: ' + type;
     },
-    EntityUpdateError: function(id, type) {
-        this.name = 'ENTITY_UPDATE_ERROR';
-        this.message = 'Error updating entity data for device: ' + id + ' of type: ' + type;
+    EntityGenericError: function(id, type, details) {
+        this.name = 'ENTITY_GENERIC_ERROR';
+        this.message = 'Error accesing entity data for device: ' + id + ' of type: ' + type;
+        this.details = details;
```

Comment:
Take into account when detail is null or undefined. Assign empty object instead.